include/color_spinor_field_order.h

```diff
@@ -247,6 +247,11 @@ namespace quda {
     {
       return parity * offset_cb + ((x_cb * nSpin + s) * nColor + c) * nVec + v;
     }
+
+    __device__ __host__ inline int wrap_index(int parity, int x_cb, int s) const
```

Comment:
Is there a reason this (and all its other incantations) is implemented as a separate function and not via default args (c = 0, v = 0) to `index`?

Reply:
Using default args is the first thing I tried, but I feel this is a clearer way to indicate that we want to access a chunk of data. This method `wrap_index` is only available for the AoS orders. Beside, function overlaoding between `const`/`non-const` and the default args makes me uncomfortable.

Reply:
Also, the default `AccessorCB` (line 229) and `GhostAccessorCB` (line 235) need error-throwing implementations of `wrap_index`, if we decide to keep `wrap_index`.

Reply:
The default ones don't have the `wrap_index` method, only the `QUDA_SPACE_SPIN_COLOR_FIELD_ORDER` and `QUDA_MILC_GAUGE_ORDER` specializations have this method. This enforces that only these order can use `wrap_index`, otherwise it's a compile time error, which is probably the thing we want?

Reply:
re: your first reply, fair point, and I guess at that rate it comes down to style---just add a comment to make it clear
re: second reply, we don't seem to have a consistent style on that in QUDA, but I agree a compile time error makes more sense